src/query/v2/plan/operator.cpp

```diff
@@ -2450,6 +2466,64 @@ class DistributedCreateExpandCursor : public Cursor {
     return edge_requests;
   }
 
+  std::vector<msgs::NewExpand> ExpandCreationInfoToRequests(MultiFrame &multi_frame, ExecutionContext &context) const {
+    std::vector<msgs::NewExpand> edge_requests;
+    auto reader = multi_frame.GetValidFramesModifier();
+
+    for (auto &frame : reader) {
+      const auto &edge_info = self_.edge_info_;
+      msgs::NewExpand request{.id = {context.edge_ids_alloc->AllocateId()}};
+      ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, nullptr,
+                                    storage::v3::View::NEW);
+      request.type = {edge_info.edge_type};
+      if (const auto *edge_info_properties = std::get_if<PropertiesMapList>(&edge_info.properties)) {
+        for (const auto &[property, value_expression] : *edge_info_properties) {
+          TypedValue val = value_expression->Accept(evaluator);
+          request.properties.emplace_back(property, storage::v3::TypedValueToValue(val));
+        }
+      } else {
+        // handle parameter
+        auto property_map = evaluator.Visit(*std::get<ParameterLookup *>(edge_info.properties)).ValueMap();
+        for (const auto &[property, value] : property_map) {
+          const auto property_id = context.request_router->NameToProperty(std::string(property));
+          request.properties.emplace_back(property_id, storage::v3::TypedValueToValue(value));
+        }
+      }
+      // src, dest
+      TypedValue &v1_value = frame[self_.input_symbol_];
+      const auto &v1 = v1_value.ValueVertex();
+      const auto &v2 = OtherVertex(frame);
+
+      // Set src and dest vertices
+      // TODO(jbajic) Currently we are only handling scenario where vertices
+      // are matched
+      const auto set_vertex = [](const auto &vertex, auto &vertex_id) {
+        vertex_id.first = vertex.PrimaryLabel();
+        vertex_id.second = vertex.GetVertex().id.second;
+      };
+
+      std::invoke([&]() {
+        switch (edge_info.direction) {
+          case EdgeAtom::Direction::IN: {
+            set_vertex(v2, request.src_vertex);
+            set_vertex(v1, request.dest_vertex);
+            break;
+          }
+          case EdgeAtom::Direction::OUT: {
+            set_vertex(v1, request.src_vertex);
+            set_vertex(v2, request.dest_vertex);
+            break;
+          }
+          case EdgeAtom::Direction::BOTH:
+            LOG_FATAL("Must indicate exact expansion direction here");
+        }
+      });
```

Comment:
That's a good question. After looking at it appears that it was assumed that there is an ordering of the `from/to` vertices of the request. However, that's not true, because the `shard_rsm` handler does not care if for two vertices v1, v2 (from = v1 or from = v2 are both valid). I removed this, good catch, as I was mostly focused on moving staff around I assumed that logic was still valid (I did not implement this cursor and my assumptions felt short).

Reply:
There is an ordering between from/to, it is purposefully called source and destination vertex: even though both are valid, but the direction of the edge is `src -> dst`, so ordering them is important.